<!DOCTYPE html>
<html>
<head><meta charset="utf-8" />
<link rel="apple-touch-icon" sizes="180x180" href="/img/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/img/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/img/favicon-16x16.png">
<link rel="manifest" href="/img/site.webmanifest">
<title>Stuff I Just Found Out</title>

<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>





<!-- Custom stylesheet, it must be in the same directory as the html file -->
<link rel="stylesheet" href="/css/notebook_preset.css">
<link rel="stylesheet" href="/css/custom.css">

<!-- Loading mathjax macro -->
<!-- Load mathjax -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML-full,Safe"> </script>
    <!-- MathJax configuration -->
    <script type="text/x-mathjax-config">
    init_mathjax = function() {
        if (window.MathJax) {
        // MathJax loaded
            MathJax.Hub.Config({
                TeX: {
                    equationNumbers: {
                    autoNumber: "AMS",
                    useLabelIds: true
                    }
                },
                tex2jax: {
                    inlineMath: [ ['$','$'], ["\\(","\\)"] ],
                    displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
                    processEscapes: true,
                    processEnvironments: true
                },
                displayAlign: 'center',
                CommonHTML: {
                    linebreaks: { 
                    automatic: true 
                    }
                },
                "HTML-CSS": {
                    linebreaks: { 
                    automatic: true 
                    }
                }
            });
        
            MathJax.Hub.Queue(["Typeset", MathJax.Hub]);
        }
    }
    init_mathjax();
    </script>
    <!-- End of mathjax configuration --></head>
<body>
<div class="panel-background">
<div class="panel-left"></div>
<div class="panel-main">

    <div class="header">
        <h1 id="title">Stuff I Just Found Out</h1>
        <p id="subtitle">A science blog with all the maths put back in.</p>
    </div> 
    <!-- Top navigation -->
    <div class="topnav">

        <!-- Centered link -->
        <div class="topnav-centered"><div>
        <a href="../index.html">Home</a>
        <a href="./about.html">About</a>
        <a href="./topics.html">Topics</a>
        <a href="./bookmarks.html">Bookmarks</a></div>
        </div>
    
    
        <!-- Right-aligned links -->
        <div class="topnav-right">
        <a href=""></a>
        </div>
    
    </div> 
    
    <div class="content">
    <div tabindex="-1" id="notebook" class="border-box-sizing">
    <div class="container" id="notebook-container">
        
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Systems-Biology-Lecture-1">Systems Biology Lecture 1<a class="anchor-link" href="#Systems-Biology-Lecture-1">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Topics-for-this-lecture">Topics for this lecture<a class="anchor-link" href="#Topics-for-this-lecture">&#182;</a></h3><ul>
<li>Michaelis-Menten kinetics</li>
<li>Activator-inhibitor curve</li>
<li>Input function</li>
<li>Coorperativity</li>
<li>Ultrasensitivity (molecular titration) work done by Nick Buchler</li>
<li>Response time ~ cell generation time</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Michaelis-Mentens-Equation">Michaelis-Mentens Equation<a class="anchor-link" href="#Michaelis-Mentens-Equation">&#182;</a></h3><p>Chemists will use [X] for concentration of X. This is very boring so we'll skip this and you can tell from context whether X is refering to X itself or the conc. of X.</p>
<p>A prototypical chemical equation for an enzyme catalysing the conversation of a substrate into a product is given by:</p>
$$ E + S \leftrightharpoons ES \rightarrow E + P $$<p>Here, these reactions have forward and backward equilibrium constants given by:</p>
<ul>
<li>$ E + S \rightarrow ES $ with rate constant $k_f$</li>
</ul>
<ul>
<li>$ ES \rightarrow E + S $ with rate constant $k_r$</li>
</ul>
<ul>
<li>$ ES \rightarrow E + P $ with rate constant $k_{cat}$</li>
</ul>
<p>This means we can set up a bunch of differential equations (using the law of mass action):</p>
<ol>
<li>$ \frac{d(ES)}{dt} = k_f \times E \times S - (k_r + k_{cat}) \times (ES)$</li>
</ol>
<ol>
<li>$ \frac{dE}{dt} = -k_f \times E \times S + (k_r + k_{cat}) \times (ES)$</li>
</ol>
<ol>
<li>$ \frac{dS}{dt} = -k_f \times E \times S + k_r \times (ES)$</li>
</ol>
<ol>
<li>$ \frac{dP}{dt} = k_{cat} \times (ES)$</li>
</ol>
<p>Here, (ES) stands for the (concentration of) enzyme-substrate complex - not the product E times S. Don't get confused by this!</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>We will also define another constant:</p>
$$K_d = k_r/k_f$$<p>called the <em>dissociation constant</em> which has dimensions in units of [Conc] (as you can see from dimensional analysis of the above equations).</p>
<ul>
<li>$\text{forward rate} = k_f[E][S]$</li>
</ul>
<ul>
<li>$\text{reverse rate} = k_r[ES]$</li>
</ul>
<p>This constant determines how much the reaction is going in the reverse direction compared to the forwards direction.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="First-thoughts-about-these-equations">First thoughts about these equations<a class="anchor-link" href="#First-thoughts-about-these-equations">&#182;</a></h4><ol>
<li>If you add equations 1 and 2 together, you get that $d[E]_{T}/dt = 0$ i.e. $[E]_T$ is constant, which is kind of obvious; enzyme does not get used up in a reaction so the total amount (in either the bound or unbound state) must be conserved. The subscript T stands for "total", and means the total concentration of that substance, both in the bound and unbound state. So for example $[S]_T = [S] + [ES]$. </li>
</ol>
<ol>
<li>The equations are nonlinear, so it would seem hard to get a closed form solution in terms of standard functions. In fact, there <em>is</em> a closed form solution in terms of the Lambert W-function, but I will leave this for later! It is a surprising fact. </li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To get some intuition, we are going to do some limits, and plot some graphs. We're also going to ignore the product-forming reaction, which means setting $k_{cat} = 0$.</p>
<p>Firstly, we want to define a fraction $ f_B = \frac{ES}{E+ES}$, the fraction of the enzyme which is bound.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Four-limit-practice-questions">Four limit practice questions<a class="anchor-link" href="#Four-limit-practice-questions">&#182;</a></h4><ol>
<li>$[S]_T \rightarrow 0$, $f_B \rightarrow 0$
Intuitively, when there is no substrate, none of the enzyme will be bound. </li>
</ol>
<ol>
<li>$[S]_T \rightarrow \infty$, $f_B \rightarrow 1$
When you have loads and loads of substrate, all the enzyme will be bound. </li>
</ol>
<ol>
<li>$[E]_T \rightarrow 0$, $f_B \rightarrow \frac{1}{1 + K_d/S}$
This is the hardest one. Imagine having a single molecule of the enzyme. Note that this is going to be proportional to the We will prove this mathematically by using L'Hopital's rule, but for now just note that the dependence on $K_d/S$ needs to be true by dimensional analysis. </li>
</ol>
<ol>
<li>$[E]_T \rightarrow \infty$, $f_B \rightarrow 0$
When you have loads of enzyme, way more than substrate, barely any of it will be bound, even if all the substrate is bound. </li>
</ol>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="REDO-THIS-BECAUSE-IT-IS-WRONG!">REDO THIS BECAUSE IT IS WRONG!<a class="anchor-link" href="#REDO-THIS-BECAUSE-IT-IS-WRONG!">&#182;</a></h4><p>A note on part 3. We are going to prove this rigourously by looking at the differential equations and then using L'Hopital's rule. However, it is also instructive to do this in another way, which may be less rigourous, but is also much faster and more broadly applicable; dimensional analysis. Imagine we only have 1 molecule of enzyme in an otherwise plentiful supply of substrate, and we want to know $f_B$, which essentially means the proportion of the time that our molecule is in the bound state. Collisions are more likely to occur when the concentration of $S$ is higher, in fact we can be more precise and say that this will be proportional to $\sqrt{S}$. (Why? Because...) So $f_B \propto \sqrt{S}$. Also note that it will certainly depend on the forewards and backwards rates, increasing with the forewards rate since that drives the binding of enzyme, and decreasing with the backwards rate. So since $K_d$ is (one over) the ratio of these quantities, and to get the dimensions to work out, we can guess that $f_B \propto \sqrt{S/K_d}$. There might be a constant of proportionality, but let's just guess that this is $1$.</p>
<p>To prove the rigourous way, let's use L'Hopital's rule, which in this case says that:</p>
$$\lim_{E\to 0}f_B = \frac{\lim_{E\to 0}ES}{\lim_{E\to 0}ES + E} = \frac{\lim_{E\to 0}d(ES)/dE}{\lim_{E\to 0}d(ES)/dE + dE/dE} $$<p>To evaluate $\frac{d(ES)}{dE}$, we can use the chain rule, "dividing" both sides by $dt$, and then using equations 1 and 2 to get $\frac{k_f \times E \times S}{k_r \times (ES)}$. We can see that our limit $L$ must satisfy the following equation:</p>
$$ L = \lim_{E \to 0}\frac{d(ES)}{dE} = \frac{k_f \times E \times S}{k_r \times (ES)} = \frac{S}{K_d \times L}$$<p>.</p>
<p>From this, we get that $L = \sqrt{S/K_d}$, and therefore that</p>
$$f_B = \frac{\sqrt{S/K_d}}{1 + \sqrt{S/K_d}} = \frac{1}{1 + \sqrt{K_d/S}}$$
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Some-maths-to-solve-part-3">Some maths to solve part 3<a class="anchor-link" href="#Some-maths-to-solve-part-3">&#182;</a></h4><p>Let's have a look at this question in a bit more detail. From equation 1, $d(ES)/dt = 0$ when $$</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Problems-with-Michaelis-Mentens">Problems with Michaelis-Mentens<a class="anchor-link" href="#Problems-with-Michaelis-Mentens">&#182;</a></h4><ol>
<li>The equation $ES \rightarrow E + P$ <em>always</em> has a reverse rate - even if very small. In MM kinetics, the reverse rate is exactly $0$, which is thermodynamically not actually possible. $P$ can <em>always</em> turn back into $S$, even if only a little bit. </li>
</ol>
<ol>
<li><p>Note - the enzyme will not change the equilibrium ratio of $S$ to $P$, it will just speed up the reaction by lowering the activation energy ($\Delta G$). No reaction will have a 100% forwards reaction! This is just saying point 1 again, but it is important to note.</p>
</li>
<li><p>So in theory MM may not be applicable in scenarios where the equilibrium is not very far to the right.</p>
</li>
</ol>
<h4 id="Velocity">Velocity<a class="anchor-link" href="#Velocity">&#182;</a></h4>$$v = \frac{S}{S + K_M}V_{max}$$<p>.</p>
<p>Here we have a new quantity, the <em>Michaelis constant</em>, $K_M$, which is defined as $\frac{k_{cat} + k_r}{k_f}$, which is basically just the dissociation constant from earlier, but now for a non-zero $k_{cat}$.</p>
<p>We also have a new quantity, $V_{max}$, which is defined as the rate of product formation at saturation of substrate, i.e. $$V_{max} = \max_{S}\frac{dP}{dt} = \lim_{S \to \infty}\frac{dP}{dt}$$.</p>
<h3 id="Simple-Models-of-Gene-Expression">Simple Models of Gene Expression<a class="anchor-link" href="#Simple-Models-of-Gene-Expression">&#182;</a></h3><h4 id="First-model-of-activation">First model of activation<a class="anchor-link" href="#First-model-of-activation">&#182;</a></h4><p>$X \rightarrow Y$</p>
<p>$X$ is a gene which produces some protein $X^*$, which binds near to the promoter of $Y$ to activate it. Given some exogenous signal, $X$ will be activated, possibly within the timescale of a second, and $X^*$ may start being produced to an equilibrium amount within the timescale of a few seconds. A lot of the time we might just forget about $X^*$, since it gets produced quickly, and we're not really interested in it per se; we only really care about the effect of the activation levels of $X$ and expression levels of $Y$, and may equivocate $X$ and $X^*$, and $Y$ and $Y^*$.</p>
<p>Let's make a model for expression levels of $Y$ (or $Y^*$) as $\dot{Y} = f(X^*) - \alpha Y$. The rate $\alpha = \alpha_{deg} + \alpha_{dil}$, i.e. this damping factor comes from degredation, and also cell growth, which acts like dilution. If we treat the function $f$ as just a constant, $\beta$, then the solution to this equation is just exponential plateauing to a non-zero value.</p>
$$\frac{d(Ye^{\alpha t})}{dt} = \beta e^{\alpha t}$$$$Y = \frac{\beta}{\alpha}  + Ce^{-\alpha t}$$<h4 id="Dependance-of-$Y$-on-$X$">Dependance of $Y$ on $X$<a class="anchor-link" href="#Dependance-of-$Y$-on-$X$">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Cooperative-Binding">Cooperative Binding<a class="anchor-link" href="#Cooperative-Binding">&#182;</a></h4><p>In our last model, we get $Y$ level depending on $X$, very much like Michaelis Mentens velocity. Can we make this a bit faster?</p>
<p>We'd like something a bit more like $f(X) = \frac{X^n}{K^n + X^n}$, which gets faster as $X$ increases. We can get this through cooperative binding. There's also an alternative, which produces ultrasensitivity without cooperativity...</p>
<p>$XW \leftrightharpoons X + W \rightarrow Y$</p>
<p>For some values of $K_W$, $K_d$, $W_T$, we can get it so that $Y$ starts being expressed only once $X$ reaches some threshold value $W_T$.</p>
<p>This is called <em>molecular titration</em>, and we'll have a look at it in the start of the next lecture.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">ipywidgets</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython3"><pre><span></span> 
</pre></div>

    </div>
</div>
</div>

</div>
    </div>
    </div>
    </div>
</div>
<div class="panel-right"></div>
</div>
</body>

 


</html>
