<!DOCTYPE html>
<html>
<head><meta charset="utf-8" />
<link rel="apple-touch-icon" sizes="180x180" href="/img/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/img/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/img/favicon-16x16.png">
<link rel="manifest" href="/img/site.webmanifest">
<title>Stuff I Just Found Out</title>

<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>





<!-- Custom stylesheet, it must be in the same directory as the html file -->
<link rel="stylesheet" href="/css/notebook_preset.css">
<link rel="stylesheet" href="/css/custom.css">

<!-- Loading mathjax macro -->
<!-- Load mathjax -->
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-MML-AM_CHTML-full,Safe"> </script>
    <!-- MathJax configuration -->
    <script type="text/x-mathjax-config">
    init_mathjax = function() {
        if (window.MathJax) {
        // MathJax loaded
            MathJax.Hub.Config({
                TeX: {
                    equationNumbers: {
                    autoNumber: "AMS",
                    useLabelIds: true
                    }
                },
                tex2jax: {
                    inlineMath: [ ['$','$'], ["\\(","\\)"] ],
                    displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
                    processEscapes: true,
                    processEnvironments: true
                },
                displayAlign: 'center',
                CommonHTML: {
                    linebreaks: { 
                    automatic: true 
                    }
                },
                "HTML-CSS": {
                    linebreaks: { 
                    automatic: true 
                    }
                }
            });
        
            MathJax.Hub.Queue(["Typeset", MathJax.Hub]);
        }
    }
    init_mathjax();
    </script>
    <!-- End of mathjax configuration --></head>
<body>
<div class="panel-background">
<div class="panel-left"></div>
<div class="panel-main">

    <div class="header">
        <h1 id="title">Stuff I Just Found Out</h1>
        <p id="subtitle">A science blog with all the maths put back in.</p>
    </div> 
    <!-- Top navigation -->
    <div class="topnav">

        <!-- Centered link -->
        <div class="topnav-centered"><div>
        <a href="/">Home</a>
        <a href="/html/about.html">About</a>
        <a href="/html/topics.html">Topics</a>
        <a href="/html/bookmarks.html">Bookmarks</a></div>
        </div>
    
    
        <!-- Right-aligned links -->
        <div class="topnav-right">
        <a href=""></a>
        </div>
    
    </div> 
    
    <div class="content">
    <div tabindex="-1" id="notebook" class="border-box-sizing">
    <div class="container" id="notebook-container">
        
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Systems-Biology-Lecture-6">Systems Biology Lecture 6<a class="anchor-link" href="#Systems-Biology-Lecture-6">&#182;</a></h1><h2 id="Feed-forward-Loop-Network-Motif">Feed-forward Loop Network Motif<a class="anchor-link" href="#Feed-forward-Loop-Network-Motif">&#182;</a></h2><h3 id="Topics">Topics<a class="anchor-link" href="#Topics">&#182;</a></h3><ul>
<li>Feed-forward loop is a network motif</li>
<li>Sign sensitive delay element (coherent type 1)</li>
<li>Pulse generator, speed-up (incoherent type 1)</li>
<li>Temporal programs</li>
</ul>
<h3 id="E.-Coli-Transcription-Network">E. Coli Transcription Network<a class="anchor-link" href="#E.-Coli-Transcription-Network">&#182;</a></h3><ul>
<li>$N$ nodes ~ 400 genes</li>
<li>$E$ edges (directed) ~ 500 interactions</li>
<li>Are there patterns that occur more regularly than you expect by chance?</li>
</ul>
<p>We have already seen that negative autoregulation appears more regularly than you would expect by chance. 
The paper talks about the concept of sparseness - there are $N^2$ possible directed edges. Let $p = \frac{E}{N^2} \ll 1$ be the probability of choosing a pair of nodes that has an edge between them. By sparseness, we just mean the observation that $p \ll 1$. Note that on average, we might expect each node to have about 1 incoming and 1 outgoing edge. However, we know from the previous lecture that some genes will have many more outgoing edges. Also note that most nodes do not code for transcription factors - and so have no outgoing edges.</p>
<p>We will try to characterise subgraphs $G$ of our graph. In particular, we will keep track of edges and nodes in the subgraph - denote these by $n$ and $e$. In particular, let's look at the feed-forward loop motif:</p>
$$ X \rightarrow Y \rightarrow Z, X \rightarrow Z $$<p>N.B. In this context, our arrows don't mean upregulation - it is just any kind of regulation (both $\rightarrow$ and $\dashv$).</p>
$$ \langle N_G \rangle = \frac{1}{a}N^np^g $$<p>This equation is the expected number of the subgraph $G$ that we should expect to see within the original graph. The factor $a$ is a symmetry factor, represented by how many ways there are to rearrange the nodes in the subgraph and get the same subgraph (automorphisms). For the repressilator it would be 3; here it is 1 as there are no symmetries of the FFL.</p>
<p>Define $\lambda = E/N$. We can then rewrite the above equation:</p>
$$ \langle N_G \rangle = \frac{1}{a}N^{n-g}\lambda^g $$<p>.</p>
<p>If $\lambda \approx 1$ then we expect to get $ \langle N_G \rangle \approx 1$ in cases where $g = n$ (like the FFL).</p>
<p>So we can ask; how many FFLs were observed in E. Coli? The answer is 42 - and the expected number is 1.7 $\pm$ 1.3. This is obviously much more than would appear by chance. Last time we talked about how the ER network is probably not a good null model, and instead we should use the degree-preserving network, which preserves the degree distribution of the whole graph. Recall the simple swapping algorithm for doing this. If we do the same analysis on a degree-preserving network, we get an expected number of FFL motifs as 7 $\pm$ 5. This is much higher than the ER network, but still much smaller than 42.</p>
<p>N.B. What experiment is done to produce these transcription networks? There are a lot of methods - using knockout strains, RNA-seq, and other ways to look at the binding affinity of each TF at each promoter site (proteonomics).</p>
<h3 id="Feed-forward-Loops">Feed-forward Loops<a class="anchor-link" href="#Feed-forward-Loops">&#182;</a></h3><h4 id="Coherent-Type-1">Coherent Type 1<a class="anchor-link" href="#Coherent-Type-1">&#182;</a></h4><p>This kind of feed-forward loop is the most common kind. 
Now $\rightarrow$ really does mean activating.</p>
<ul>
<li>$X \rightarrow Y \rightarrow Z \text{ and } X \rightarrow Z $</li>
<li>the and is a logical and - activating $Z$ requires input from both $X$ and $Y$. </li>
<li>$X$ and $Y$ are also activated by some exogenous signals $S_X$ and $S_Y$. </li>
<li>this circuit causes a sign-sensitive delay</li>
</ul>
<p>Suppose:</p>
<ul>
<li>$S_Y$ is present</li>
<li>$S_X$ switches from off to on</li>
</ul>
<p><strong>Draw a graph!</strong></p>
<p>Then $Z$ will be activated after a small delay by $S_X$. However, when $S_X$ switches back off, there is no delay, and $Z$ is switched off immediately. 
If the and gate was changed to an or gate, then there wouldn't be an on-delay, but there would be an off-delay.</p>
<p>Now suppose:</p>
<ul>
<li>$S_X$ is present</li>
<li>$S_Y$ switches from on to off</li>
</ul>
<p>In this case, there isn't any kind of delay to $Z$; this is just equivalent to $Y$ regulating $Z$.</p>
<h4 id="Incoherent-Type-1">Incoherent Type 1<a class="anchor-link" href="#Incoherent-Type-1">&#182;</a></h4><ul>
<li>$X \rightarrow Y \dashv Z \text{ and } X \rightarrow Z$</li>
</ul>
<p>We will use the boolean approximation:</p>
<table>
<thead><tr>
<th>X*</th>
<th>Y*</th>
<th>expression of Z</th>
</tr>
</thead>
<tbody>
<tr>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr>
<td>0</td>
<td>1</td>
<td>0</td>
</tr>
<tr>
<td>1</td>
<td>0</td>
<td>1</td>
</tr>
<tr>
<td>1</td>
<td>1</td>
<td>0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><strong>Draw a graph!</strong></p>
<p>What we end up with is a sort of pulse of $Z$ where expression starts off high but is then quickly repressed by $Y$. We can also think of this as saying that the on-time is shortened. (N.B. We often define $t_{on}$ the on-time, as the time taken to get half of the long-term equilibrium expression concentration, and compare this just to simple repression.)</p>
<p>We can now make a table of a few strategies we have seen to decrease the on and off times:</p>
<table>
<thead><tr>
<th>Method</th>
<th>$t_{on}$</th>
<th>$t_{off}$</th>
</tr>
</thead>
<tbody>
<tr>
<td>Increase degredation rate</td>
<td>dec.</td>
<td>dec.</td>
</tr>
<tr>
<td>Negative autoregulation</td>
<td>dec.</td>
<td>nothing</td>
</tr>
<tr>
<td>Incoherent type 1 FFL</td>
<td>dec.</td>
<td>nothing</td>
</tr>
</tbody>
</table>
<p>N.B. Transcription is slow! These timescale may be on the order of 10s of minutes. To be fast, you cannot wait for protein concentrations to change - you need to change protein state, with e.g. phosphorilation and kinase cascades. Processes like changing states of proteins (things like $S_X$ binding $X$) happen on the order of less than a second and are fast.</p>
<h4 id="Temporal-Programs-and-the-Single-Input-Module">Temporal Programs and the Single Input Module<a class="anchor-link" href="#Temporal-Programs-and-the-Single-Input-Module">&#182;</a></h4><p>Suppose we have $Z_1: p_2 \rightarrow p_2$, $Z_2: p_2 \rightarrow p_3... $ and so on - i.e. a common pattern in a metabolic pathway. It would make sense for the cell to synthesize $Z_1$ first, then $Z_2$, etc.</p>
<p>The <em>single input module</em> does exactly this:</p>
<ul>
<li>$X \rightarrow Z_1, Z_2, Z_3, Z_4...$, $X \rightarrow X$. </li>
<li>one transcription factor activates many input proteins - often when all of the $Z$'s are in the same input pathway. </li>
<li>We can do this s.t. the $Z$'s are produced in the correct order. As long as the $Z_i$ have different activation thresholds $k_i$ that are increasing, then this will work. </li>
</ul>
<p>Question:</p>
<ul>
<li>Once the source food $p_1$ goes away, what happens to the order of degredation our metabolic enzymes? Is it FIFO or LIFO? The single input module is LIFO - in a biosynthetic pathway we really want a FIFO queue because otherwise we will end up with a build up of intermediate metabolites. </li>
</ul>
<p>In the E. Coli flagella production queue, we find that this is in fact a FIFO queue. How might we be able to do this? The answer is a multi-output FFL:</p>
<ul>
<li>$X \rightarrow Y \rightarrow Z_1, Z_2, ..., Z_n \text{ or } X \rightarrow Z_1, Z_2, ... Z_n$. </li>
</ul>
<ul>
<li>We have two sets of constants $k_i$ controlling the activations $X \rightarrow Z_i$ and $k'_i$ controlling $Y \rightarrow Z_i$. </li>
</ul>
<ul>
<li>The constants are ordered: $k_1 &lt; k_2 &lt; ... &lt; k_n$ and $k'_1 &gt; k'_2 &gt; ... &gt; k'_n$.</li>
</ul>
<ul>
<li>This works because $Y$ gets delayed on and off relative to $X$ (with some other constant $K$), and because we have an or-gate, $X$ determintes the on-time of the pathway enzymes and $Y$ determines the off-times. </li>
</ul>

</div>
</div>
</div>
    </div>
    </div>
    </div>
</div>
<div class="panel-right"></div>
</div>
</body>

 


</html>
