{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "following-exemption",
   "metadata": {},
   "source": [
    "# Systems Biology Lecture 6\n",
    "\n",
    "## Feed-forward Loop Network Motif\n",
    "\n",
    "### Topics\n",
    "\n",
    "- Feed-forward loop is a network motif\n",
    "- Sign sensitive delay element (coherent type 1)\n",
    "- Pulse generator, speed-up (incoherent type 1)\n",
    "- Temporal programs\n",
    "\n",
    "### E. Coli Transcription Network\n",
    "\n",
    "- $N$ nodes ~ 400 genes\n",
    "- $E$ edges (directed) ~ 500 interactions\n",
    "- Are there patterns that occur more regularly than you expect by chance?\n",
    "\n",
    "We have already seen that negative autoregulation appears more regularly than you would expect by chance. \n",
    "The paper talks about the concept of sparseness - there are $N^2$ possible directed edges. Let $p = \\frac{E}{N^2} \\ll 1$ be the probability of choosing a pair of nodes that has an edge between them. By sparseness, we just mean the observation that $p \\ll 1$. Note that on average, we might expect each node to have about 1 incoming and 1 outgoing edge. However, we know from the previous lecture that some genes will have many more outgoing edges. Also note that most nodes do not code for transcription factors - and so have no outgoing edges. \n",
    "\n",
    "\n",
    "We will try to characterise subgraphs $G$ of our graph. In particular, we will keep track of edges and nodes in the subgraph - denote these by $n$ and $e$. In particular, let's look at the feed-forward loop motif:\n",
    "\n",
    "$$ X \\rightarrow Y \\rightarrow Z, X \\rightarrow Z $$\n",
    "\n",
    "\n",
    "N.B. In this context, our arrows don't mean upregulation - it is just any kind of regulation (both $\\rightarrow$ and $\\dashv$). \n",
    "\n",
    "\n",
    "$$ \\langle N_G \\rangle = \\frac{1}{a}N^np^g $$\n",
    "\n",
    "This equation is the expected number of the subgraph $G$ that we should expect to see within the original graph. The factor $a$ is a symmetry factor, represented by how many ways there are to rearrange the nodes in the subgraph and get the same subgraph (automorphisms). For the repressilator it would be 3; here it is 1 as there are no symmetries of the FFL. \n",
    "\n",
    "Define $\\lambda = E/N$. We can then rewrite the above equation:\n",
    "\n",
    "$$ \\langle N_G \\rangle = \\frac{1}{a}N^{n-g}\\lambda^g $$. \n",
    "\n",
    "If $\\lambda \\approx 1$ then we expect to get $ \\langle N_G \\rangle \\approx 1$ in cases where $g = n$ (like the FFL). \n",
    "\n",
    "So we can ask; how many FFLs were observed in E. Coli? The answer is 42 - and the expected number is 1.7 $\\pm$ 1.3. This is obviously much more than would appear by chance. Last time we talked about how the ER network is probably not a good null model, and instead we should use the degree-preserving network, which preserves the degree distribution of the whole graph. Recall the simple swapping algorithm for doing this. If we do the same analysis on a degree-preserving network, we get an expected number of FFL motifs as 7 $\\pm$ 5. This is much higher than the ER network, but still much smaller than 42. \n",
    "\n",
    "\n",
    "\n",
    "N.B. What experiment is done to produce these transcription networks? There are a lot of methods - using knockout strains, RNA-seq, and other ways to look at the binding affinity of each TF at each promoter site (proteonomics). \n",
    "\n",
    "### Feed-forward Loops\n",
    "\n",
    "#### Coherent Type 1\n",
    "\n",
    "This kind of feed-forward loop is the most common kind. \n",
    "Now $\\rightarrow$ really does mean activating. \n",
    "\n",
    "- $X \\rightarrow Y \\rightarrow Z \\text{ and } X \\rightarrow Z $\n",
    "- the and is a logical and - activating $Z$ requires input from both $X$ and $Y$. \n",
    "- $X$ and $Y$ are also activated by some exogenous signals $S_X$ and $S_Y$. \n",
    "- this circuit causes a sign-sensitive delay\n",
    "\n",
    "Suppose:\n",
    "\n",
    "- $S_Y$ is present\n",
    "- $S_X$ switches from off to on\n",
    "\n",
    "**Draw a graph!**\n",
    "\n",
    "Then $Z$ will be activated after a small delay by $S_X$. However, when $S_X$ switches back off, there is no delay, and $Z$ is switched off immediately. \n",
    "If the and gate was changed to an or gate, then there wouldn't be an on-delay, but there would be an off-delay. \n",
    "\n",
    "\n",
    "Now suppose:\n",
    "\n",
    "- $S_X$ is present\n",
    "- $S_Y$ switches from on to off\n",
    "\n",
    "In this case, there isn't any kind of delay to $Z$; this is just equivalent to $Y$ regulating $Z$. \n",
    "\n",
    "#### Incoherent Type 1\n",
    "\n",
    "- $X \\rightarrow Y \\dashv Z \\text{ and } X \\rightarrow Z$\n",
    "\n",
    "We will use the boolean approximation:\n",
    "\n",
    "| X* | Y* | expression of Z |\n",
    "|----|----|-----------------|\n",
    "| 0 | 0 | 0 |\n",
    "| 0 | 1 | 0 |\n",
    "| 1 | 0 | 1 |\n",
    "| 1 | 1 | 0 |"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "artistic-storm",
   "metadata": {},
   "source": [
    "**Draw a graph!**\n",
    "\n",
    "What we end up with is a sort of pulse of $Z$ where expression starts off high but is then quickly repressed by $Y$. We can also think of this as saying that the on-time is shortened. (N.B. We often define $t_{on}$ the on-time, as the time taken to get half of the long-term equilibrium expression concentration, and compare this just to simple repression.)\n",
    "\n",
    "\n",
    "We can now make a table of a few strategies we have seen to decrease the on and off times:\n",
    "\n",
    "| Method | $t_{on}$ | $t_{off}$ |\n",
    "|--------|----------|-----------|\n",
    "|Increase degredation rate | dec. | dec. |\n",
    "| Negative autoregulation | dec. | nothing |\n",
    "| Incoherent type 1 FFL | dec. | nothing |\n",
    "\n",
    "N.B. Transcription is slow! These timescale may be on the order of 10s of minutes. To be fast, you cannot wait for protein concentrations to change - you need to change protein state, with e.g. phosphorilation and kinase cascades. Processes like changing states of proteins (things like $S_X$ binding $X$) happen on the order of less than a second and are fast. \n",
    "\n",
    "\n",
    "#### Temporal Programs and the Single Input Module\n",
    "\n",
    "Suppose we have $Z_1: p_2 \\rightarrow p_2$, $Z_2: p_2 \\rightarrow p_3... $ and so on - i.e. a common pattern in a metabolic pathway. It would make sense for the cell to synthesize $Z_1$ first, then $Z_2$, etc. \n",
    "\n",
    "The *single input module* does exactly this:\n",
    "\n",
    "- $X \\rightarrow Z_1, Z_2, Z_3, Z_4...$, $X \\rightarrow X$. \n",
    "- one transcription factor activates many input proteins - often when all of the $Z$'s are in the same input pathway. \n",
    "- We can do this s.t. the $Z$'s are produced in the correct order. As long as the $Z_i$ have different activation thresholds $k_i$ that are increasing, then this will work. \n",
    "\n",
    "Question:\n",
    "- Once the source food $p_1$ goes away, what happens to the order of degredation our metabolic enzymes? Is it FIFO or LIFO? The single input module is LIFO - in a biosynthetic pathway we really want a FIFO queue because otherwise we will end up with a build up of intermediate metabolites. \n",
    "\n",
    "In the E. Coli flagella production queue, we find that this is in fact a FIFO queue. How might we be able to do this? The answer is a multi-output FFL:\n",
    "\n",
    "\n",
    "- $X \\rightarrow Y \\rightarrow Z_1, Z_2, ..., Z_n \\text{ or } X \\rightarrow Z_1, Z_2, ... Z_n$. \n",
    "\n",
    "\n",
    "- We have two sets of constants $k_i$ controlling the activations $X \\rightarrow Z_i$ and $k'_i$ controlling $Y \\rightarrow Z_i$. \n",
    "\n",
    "\n",
    "- The constants are ordered: $k_1 < k_2 < ... < k_n$ and $k'_1 > k'_2 > ... > k'_n$.\n",
    "\n",
    "\n",
    "- This works because $Y$ gets delayed on and off relative to $X$ (with some other constant $K$), and because we have an or-gate, $X$ determintes the on-time of the pathway enzymes and $Y$ determines the off-times. "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "blog",
   "language": "python",
   "name": "blog"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
